Library/Homebrew/dependency_collector.rb

```diff
@@ -72,6 +72,14 @@ def xz_dep_if_needed(tags)
 
   def ld64_dep_if_needed(*); end
 
+  def zip_dep_if_needed(tags)
+    Dependency.new("zip", tags)
```

Comment:
Use `which` to check first whether `zip` is provided by the host, and make this dependency conditional.

Reply:
Any preference as to which one to use:
```
Dependency.new("zip", tags) unless which("zip")
```
or 
```
return if which("zip")
Dependency.new("zip", tags)
```
?

Reply:
I prefer the former.

Reply:
Ok, done.